### PR TITLE
Save VAT on db on plugin activation

### LINKED ISSUE #200: Save and get VAT
## Get vat from db when it will be implemented.

https://github.com/CTMobi/Wubtitle-Wordpress-Plugin/blob/61b0d4b203bebac6ea0071958ec579cd18e57cd8/src/stripe/Components/InfoPriceColumn.js#L9-L14

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 61b0d4b203bebac6ea0071958ec579cd18e57cd8 in #197. cc @CTMobi.

_Originally posted by @todo-ctmobi in https://github.com/CTMobi/Wubtitle-Wordpress-Plugin/pull/197#issuecomment-661760538_

### PR DESCRIPTION
**Pull request by nicola-palermo**
### All Submissions:

*   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
*   [x] Does your code has proper inline documentation? 

### Changes proposed in this Pull Request:

Save VAT info on db and display them on upgrade plan form.

Closes #200  .

### Other information:

*   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
*   [ ] Have you written new tests for your changes, as applicable?
*   [ ] Have you successfully run tests with your changes locally?

### Changelog entry
Save VAT info on db and display them on upgrade plan form.

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- includes/Api/ApiPricingPlan.php  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388&bid=19275884)

File: assets/css/stripeStyle.css [hunk 1]
```diff
@@ -269,3 +269,17 @@ table tr {
...
+}
+
+.cut-line {
+	width: 59px;
+	height: 2px;
+	background-color: #ffd600;
+	display: inline-block;
+    margin-left: -58px;
+    position: absolute;
+    transform: rotate(-18deg) translateY(8px);
```
Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:284]: Codacy found an issue: [Expected indentation of 1 tab (indentation)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/stripe/Components/InfoPriceColumn.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388&bid=19275884)

File: assets/css/stripeStyle.css [hunk 2]
```diff
@@ -269,3 +269,17 @@ table tr {
...
+.cut-vat {
+	margin-right: 1rem;
+}
+
+.cut-line {
+	width: 59px;
+	height: 2px;
+	background-color: #ffd600;
+	display: inline-block;
+    margin-left: -58px;
```
Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:282]: Codacy found an issue: [Expected indentation of 1 tab (indentation)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388)

File: assets/css/stripeStyle.css [hunk 3]
```diff
@@ -269,3 +269,17 @@ table tr {
...
+    position: absolute;
```
Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:283]: Codacy found an issue: [Expected indentation of 1 tab (indentation)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388)

File: includes/Core/Cron.php [hunk 4]
```diff
@@ -88,16 +88,21 @@ public function get_remote_data() {
...
 					'totalJobs'    => $plan->totalJobs,
 					'totalSeconds' => $plan->totalSeconds,
 					// phpcs:enable
-					'price'        => $plan->price,
-					'dot_list'     => $plan->dotlist,
-					'icon'         => $plan->icon,
+					'price'          => $plan->price,
+					'dot_list'       => $plan->dotlist,
+					'icon'           => $plan->icon,
+					// phpcs:disable
```
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:101]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php [hunk 5]
```diff
@@ -88,16 +88,21 @@ public function get_remote_data() {
...
-					'dot_list'     => $plan->dotlist,
-					'icon'         => $plan->icon,
+					'price'          => $plan->price,
+					'dot_list'       => $plan->dotlist,
+					'icon'           => $plan->icon,
+					// phpcs:disable 
+					// warning camel case
+					'tax_percentage' => $plan->taxPercentage,
+					'tax_amount'     => $plan->taxAmount,
+					// phpcs:enable
```
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:105]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:101]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:105]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:101]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:105]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:101]: I need it, I get these camel case values from backend endpoint.

File: includes/Core/Cron.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicola-palermo) [reply includes/Core/Cron.php:105]: I need it, I get these camel case values from backend endpoint.

File: src/stripe/Components/InfoPriceColumn.js [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
-	const vat = ((price / 100) * vatPer).toFixed(2);
-	const total = parseFloat(price) + parseFloat(vat);
+	const { price, name, taxPercentage, taxAmount } = props;
+	const total = parseFloat(price) + parseFloat(taxAmount);
+
+	let cutVat = false;
+	if (taxPercentage === 0) {
+		cutVat = ((price / 100) * 22).toFixed(2);
+	}
+
```
Comment (HyprCoder) [inline src/stripe/Components/InfoPriceColumn.js:12]: When working with currencies, it's better to use some library already tested, like this [one](https://github.com/dinerojs/dinero.js)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/stripe/Components/InfoPriceColumn.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388&bid=19275884)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/stripe/Components/InfoPriceColumn.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388&bid=19275884)

File: src/stripe/Components/InfoPriceColumn.js (hunk 6 above)
Comment (nicola-palermo) [reply src/stripe/Components/InfoPriceColumn.js]: I used the value from db.

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/stripe/Components/InfoPriceColumn.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5888388&bid=19275884)

**Pull request closed without merging**